src/RectangleA.java

```diff
@@ -0,0 +1,250 @@
+     */
+    public String toString() {
+        return "Width=" + _width + " Height=" + _height + " PointSW=" + _pointSW.toString();
+    }
+
+    /**
+     * Calculates the perimeter of the rectangle
+     *
+     * @return The perimeter of the rectangle.
+     */
+    public int getPerimeter() {
+        return 2 * _width + 2 * _height;
+    }
+
+    /**
+     * Calculates the area of the rectangle
+     *
+     * @return The area of the rectangle.
+     */
+    public int getArea() {
+        return _width * _height;
+    }
+
+    /**
+     * Move the rectangle by deltaX in x direction and deltaY in y direction
+     *
+     * @param deltaX - translate the rectangle deltaX in the x direction.
+     * @param deltaY - translate the rectangle deltaY in the y direction.
+     */
+    public void move(int deltaX, int deltaY) {
+        _pointSW.move(deltaX, deltaY);
+    }
+
+    /**
+     * Returns true if the given rectangle is equal to this rectangle
+     *
+     * @param other - the rectangle to check equality with
+     * @return true if other and this rectangle are equal
+     */
+    //TODO: NOT SURE!
+    public boolean equals(RectangleA other) {
+        return this.toString().equals(other.toString());
+    }
+
+    /**
+     * Returns the length of the rectangle diagonal
+     *
+     * @return the length of the diagonal of the Rectangle
+     */
+    public double getDiagonalLength() {
+        return _pointSW.distance(getPointNE());
+    }
+
+    /**
+     * Returns true if the current rectangle is larger than the parameter rectangle
+     *
+     * @param other - another Rectangle to compare with
+     * @return true - if the current Rectangle's area is larger than the other Rectangle which received as parameter, otherwise -false
+     */
+    public boolean isLarger(RectangleA other) {
+        return this.getArea() > other.getArea();
+    }
+
+    /**
+     * Returns the North-East point of the rectangle
+     *
+     * @return a copy of the North-East point of the Rectangle
+     */
+    public Point getPointNE() {
+        return new Point(_pointSW.getX() + _width, _pointSW.getY() + _height);
+    }
+
+    /**
+     * Changes the width to height and vice versa
+     */
+    // TODO: NOT SURE!
+    public void changeSides() {
+        int temp = _width;
+        _width = _height;
+        _height = temp;
+    }
+
+    /**
+     * Returns true if the current rectangle is in the parameter rectangle
+     *
+     * @param r - another Rectangle to check with
+     * @return true - if the current Rectangle's completely in the other Rectangle which received as parameter, otherwise -false
+     */
+    public boolean isIn(RectangleA r) {
+        // all the options that rectangle r is PARTIALLY OUT of this rectangle. So return not any of these options.
+        // TODO: Why tho or way is not working? - tester fails with the or. And is working good
```

Comment:
My proposal is to break the return statement into small conditions.

Example:
`return (A && !B) || C` is complicated.

This is easier to understand:
```
if (C) { return true }
if (!A) { return false }
if (B) { return false}
return true
```